lua/frecency/picker.lua

```diff
@@ -0,0 +1,264 @@
+---@param filter string
+---@return boolean
+---@TODO: make it more readable please :D
+M.update_results = function(filter)
+  local filter_updated = false
+  local ws_dir = filter and M.config.user_workspaces[filter]
+
+  if filter == "LSP" and not vim.tbl_isempty(M.lsp_workspaces) then
+    ws_dir = M.lsp_workspaces[1]
+  elseif filter == "CWD" then
+    ws_dir = M.cwd
+  end
+
+  if ws_dir ~= M.active_filter then
+    filter_updated = true
+    M.active_filter, M.active_filter_tag = ws_dir, filter
+  end
+
+  if vim.tbl_isempty(M.results) or filter_updated then
+    M.results = M.db.get_files { ws_dir = ws_dir, show_unindexed = M.config.show_unindexed, with_score = true }
+  end
+  return filter_updated
+end
+
+---Create entry maker function.
+---@param entry table
+---@return function
+---FIXME: path transacted with icons being pre-appended instead of appended
+M.maker = function(entry)
+  local items = {}
+  if M.config.show_scores then
+    table.insert(items, { width = 8 })
+  end
+
+  if M.config.show_filter_column then
+    local width = 0
+    -- TODO: Only add +1 if M.show_filter_thing is true, +1 is for the trailing slash
+    if M.active_filter and M.active_filter_tag == "LSP" then
+      width = #(utils.path_tail(M.active_filter)) + 1
+    elseif M.active_filter then
+      width = #(Path:new(M.active_filter):make_relative(os_home)) + 1
+    end
+    table.insert(items, { width = width })
+  end
+
+  if not M.config.disable_devicons and has_devicons then
+    table.insert(items, { width = 2 }) -- icon column
+  end
+
+  table.insert(items, { remaining = true })
+
+  return {
+    filename = entry.path,
+    ordinal = entry.path,
+    name = entry.path,
+    score = entry.score,
+    display = function(e)
+      return entry_display.create {
+        separator = "",
+        items = items,
+        hl_chars = { [os_path_sep] = "TelescopePathSeparator" },
+      }(M.display_items(e))
+    end,
+  }
+end
+
+M.display_items = function(e)
+  local items = M.config.show_scores and { { e.score, "TelescopeFrecencyScores" } } or {}
+
+  -- TODO: store the column lengths here, rather than recalculating in get_display_cols()
+  -- TODO: only include filter_paths column if M.show_filter_col is true
+
+  table.insert(items, {
+    (function()
+      if M.active_filter_tag == "LSP" or M.active_filter_tag == "CWD" then
+        return ts_util.path_tail(M.active_filter) .. os_path_sep
+      elseif M.active_filter then
+        return Path:new(M.active_filter):make_relative(os_home) .. os_path_sep
+      else
+        return ""
+      end
+    end)(),
+    "Directory",
+  })
+
+  table.insert(items, {
+    M.file_format(e.name),
+    util.buf_is_loaded(e.name) and "TelescopeBufferLoaded" or "",
+  })
+
+  if has_devicons and not M.config.disable_devicons then
+    icon, icon_highlight = devicons.get_icon(e.name, string.match(e.name, "%a+$"), { default = true })
+    table.insert(items, { icon, icon_highlight })
+  end
```

Comment:
The same can kinda be said about devicons 

I already proposed that to sunjon but he didn't wanted to integrate it or more he didnt even react. :laughing: Here is the snippet that i send him:

```lua
local utils = require "telescope.utils"

local opts = {}
-- opts.disable_devicons = true
-- opts.path_display = { "shorten" }
-- opts.path_display = { shorten = 2 }
-- opts.path_display = { "tail" }
-- opts.path_display = function(opts, path)
--   local tail = utils.path_tail(path)
--   return string.format("%s (%s)", tail, path)
-- end

local entry = { value = "lua/telescope/builtin/init.lua" }
local display = utils.transform_path(opts, entry.value)
local hl_group
display, hl_group = utils.transform_devicons(entry.value, display, opts.disable_devicons)
print(display, hl_group)
```

Reply:
Also relevant: https://github.com/nvim-telescope/telescope.nvim/pull/1139

(I don't think there's a global `disable_devicons`, though I'd like one ;))

Reply:
Would be nice to have it globally, although probably a string option makes more sense (since `enabled = false` and `color = true` makes no sense ;))

Reply:
maybe it wasnt the best idea. I am just saying we already have a `color_devicons = true|false` i global maybe we can handle devicons setup in one table. If not we can add `disable_devicons = true|false` just to the global config table as well.